src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs

```diff
@@ -313,20 +313,34 @@ private bool CanSkipAccountPicker(Authority authority)
             }
         }
 
-        private string WorkaroundOrganizationsBug(AuthenticationRequestParameters authenticationRequestParameters, WebAccount wamAccount)
+        private async Task<string> WorkaroundOrganizationsBugAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            WebAccountProvider webAccountProvider)
         {
             string differentAuthority = null;
-            if (!_wamOptions.MsaPassthrough &&
```

Comment:
Previous filter:
Not MSA-PT **AND** Organizations request **AND** (Organizations provider **OR** Any Default OS account)

Current filter:
Organizations request **AND** (Organizations provider **OR** Default OS AAD account)

Are we missing the inclusion of Default OS MSA account in the filter?

Also in which cases is organizations request is true but organizations provider is false? Is it when Default OS account is used? Provider is default then?

Reply:
MSA accounts are not affected by this bug. Previously I did not test with this setup (machine with only 1 MSA account on it).

Reply:
`Also in which cases is organizations request is true but organizations provider is false? Is it when Default OS account is used? Provider is default then?`

In MSA-PT scenario, the authority is env/organizations, and account provider can be MSA.